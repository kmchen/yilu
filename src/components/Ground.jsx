import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import Avatar from '@material-ui/core/Avatar';
import Typography from '@material-ui/core/Typography';
import { red } from '@material-ui/core/colors';
import DepartureBoard from '@material-ui/icons/DepartureBoard';
import Grid from '@material-ui/core/Grid';
import Box from '@material-ui/core/Box';

const useStyles = makeStyles(() => ({
  avatar: {
    backgroundColor: red[500],
  },
}));

export default function Ground({ address }) {
  const classes = useStyles();
  return (
    <Grid container alignItems="center" spacing={1}>
      <Grid item xs={2}>
        <Avatar className={classes.avatar}>
          <DepartureBoard />
        </Avatar>
      </Grid>
      <Grid item xs={10}>
        <Typography>Pick up from</Typography>
        <Box color="text.primary">{address}</Box>
      </Grid>
    </Grid>
  );
}

/* eslint-disable react/forbid-prop-types */
Ground.propTypes = {
  address: PropTypes.string.isRequired,
};
