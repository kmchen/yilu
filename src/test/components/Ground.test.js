import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Ground from '../../components/Ground';
import data from '../../../data/data.js';

configure({ adapter: new Adapter() });

describe('Ground component', () => {
  let groundComponent;

  beforeEach(() => {
    const props = {
      address: 'some address',
    };
    groundComponent = shallow(<Ground {...props} />);
  });

  it('should render Ground component', () => {
    expect(groundComponent).toMatchSnapshot();
  });
});

