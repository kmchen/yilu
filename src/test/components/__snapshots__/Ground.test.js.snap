// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ground component should render Ground component 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  container={true}
  spacing={1}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={2}
  >
    <WithStyles(ForwardRef(Avatar))
      className="makeStyles-avatar-1"
    >
      <DepartureBoardIcon />
    </WithStyles(ForwardRef(Avatar))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={10}
  >
    <WithStyles(ForwardRef(Typography))>
      Pick up from
    </WithStyles(ForwardRef(Typography))>
    <Styled(MuiBox)
      color="text.primary"
    >
      some address
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
